system/TestBox.cfc

```diff
@@ -239,36 +241,42 @@ component accessors="true"{
 
 		coverageService.beginCapture();
 
-		// iterate and run the test bundles
-		for( var thisBundlePath in variables.bundles ){
-			// Skip interfaces, they are not testable
-			if( getComponentMetadata( thisBundlePath ).type eq "interface" ){
-				continue;
-			}
-			// Execute Bundle
-			testBundle(
-				bundlePath  = thisBundlePath,
-				testResults = results,
-				callbacks   = arguments.callbacks
-			);
-
-			// Eager Failures on Bundle?
-			if( arguments.eagerFailure ){
-				var failuresDetected = results.getBundleStats()
-					// Get stats for running bundle
-					.filter( function( item ){
-						return ( item.path == thisBundlePath ? true : false );
-					} )
-					.reduce( function( result, item ){
-						return ( item.totalError + item.totalFail ) > 0;
-					}, false );
-
-				if( failuresDetected ){
-					// Hard skip iterations
-					break;
-				}
-			}
-		}
+        // iterate and run the test bundles
+        var threadNames = [];
+        for ( var thisBundlePath in variables.bundles ) {
+            if ( arguments.asyncBundles ) {
+                var thisThreadName = getUtility().slugify( "tb-" & thisBundlePath & "-#hash( getTickCount() + randRange( 1, 10000000 ) )#" );
+                arrayAppend( threadNames, thisThreadName );
+                thread
+                    name = thisThreadName
+                    threadName = thisThreadName
+                    thisBundlePath = thisBundlePath
+                    results = results
```

Comment:
This file will be problematic if we pass into the thread like this, because in Adobe it will be duplicated and not by reference. Meaning that each thread will get a different copy of the results.

To avoid this, place the `results` in the `request` scope and then reference it via the scoped reference instead.